benchmarks/microbenchmarks/_plan.py

```diff
@@ -0,0 +1,59 @@
+import itertools
+
+from microbenchmarks._result import Result
+from microbenchmarks._utils import get_ti_arch, tags2name
+
+import taichi as ti
+
+
+class BenchmarkPlan:
+    def __init__(self, name='plan', arch='x64', basic_repeat_times=1):
+        self.name = name
+        self.arch = arch
+        self.basic_repeat_times = basic_repeat_times
+        self.info = {'name': self.name}
+        self.plan = {}  # {'tags': [...], 'result': None}
+        self.items = []
+        self.func = None
+
+    def create_plan(self, *items):
+        items_list = [[self.name]]
+        for item in items:
+            self.items.append(item())
+            items_list.append(item().get_tags())
+            self.info[item.name] = item().get_tags()
+        case_list = list(itertools.product(*items_list))  #items generate case
+        for tags in case_list:
+            self.plan[tags2name(tags)] = {'tags': tags, 'result': None}
+
+    def set_func(self, func):
+        self.func = func
+
+    def run(self):
+        for case, plan in self.plan.items():
+            tag_list = plan['tags']
+            self._init_taichi(self.arch, tag_list)
+            _ms = self._run_func(tag_list)
+            plan['result'] = _ms
+            print(f'{tag_list}={_ms}')
+            ti.reset()
+        rdict = {'results': self.plan, 'info': self.info}
+        return rdict
+
+    def _get_kwargs(self, tags):
+        kwargs = {}
+        for item in self.items:
```

Comment:
If the order of `items` and `tags` are the same, maybe we can iterate through `zip(self.items, tags)` directly?

Reply:
bf54c75, fixed by removing case_name from the tag list, thanks! 
```
tags = tags[1:] 
for item, tag in zip(self.items, tags):
```